src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-configuration-page/rgw-configuration-page.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+
+import { NgbActiveModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import _ from 'lodash';
+
+import { Permissions } from '~/app/shared/models/permissions';
+
+import { RgwBucketService } from '~/app/shared/api/rgw-bucket.service';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { ModalService } from '~/app/shared/services/modal.service';
+import { RgwConfigModalComponent } from '../rgw-config-modal/rgw-config-modal.component';
+import { RgwBucketEncryptionModel } from '../models/rgw-bucket-encryption';
+
+@Component({
+  selector: 'cd-rgw-configuration-page',
+  templateUrl: './rgw-configuration-page.component.html',
+  styleUrls: ['./rgw-configuration-page.component.scss'],
+  providers: [RgwBucketEncryptionModel]
+})
+export class RgwConfigurationPageComponent extends ListWithDetails implements OnInit {
+  readonly vaultAddress = /^((https?:\/\/)|(www.))(?:([a-zA-Z]+)|(\d+\.\d+.\d+.\d+)):\d{4}$/;
+
+  kmsProviders: string[];
+
+  columns: Array<CdTableColumn> = [];
+
+  configForm: CdFormGroup;
+  permissions: Permissions;
+  encryptionConfigValues: any = [];
+  selection: CdTableSelection = new CdTableSelection();
+
+  @Output()
+  submitAction = new EventEmitter();
+  authMethods: string[];
+  secretEngines: string[];
+  tableActions: CdTableAction[];
+  bsModalRef: NgbModalRef;
+  filteredEncryptionConfigValues: {};
+  excludeProps: any[] = [];
+  disableCreate = false;
+  allEncryptionValues: any;
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwBucketService: RgwBucketService,
+    public authStorageService: AuthStorageService,
+    private rgwEncryptionModal: RgwBucketEncryptionModel,
```

Comment:
I was wondering why you are invoking this in the constructor. And I realized that this should have been an `interface` rather than a `class` which is wrong :face_with_diagonal_mouth: